src/dev-spaces-preview/azext_dev_spaces_preview/custom.py

```diff
@@ -38,33 +38,35 @@ def ads_use_dev_spaces(cluster_name, resource_group_name, space_name='default',
 
     from subprocess import PIPE
     should_create_resource = False
+    create_resource_ret_code = 0
     retCode = subprocess.call(
         [azds_cli, 'resource', 'select', '-n', cluster_name, '-g', resource_group_name],
         stderr=PIPE)
     if retCode == 1:
         should_create_resource = True
+        create_resource_ret_code = 1
 
     if should_create_resource:
-        retCode = subprocess.call(
+        create_resource_ret_code = subprocess.call(
             [azds_cli, 'resource', 'create', '--aks-name', cluster_name, '--aks-resource-group',
              resource_group_name, '--name', cluster_name, '--resource-group', resource_group_name],
             universal_newlines=True)
```

Comment:
this removes the need for `should_create_resource` and later check for `create_resource_ret_code`
```
if retCode == 1:
    create_resource_ret_code = ....
    if create_resource_ret_code == 1:
        return
```

Reply:
Agree